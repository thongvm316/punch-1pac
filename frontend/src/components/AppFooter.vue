<template>
  <footer class="footer">
    <div class="container grid-xl">
      <div class="footer-items">
        <ul class="list-inline">
          <li><img src="../assets/logo.png"></li>
          <li><a :href="`${meta.base_url}/terms`">{{ $t('footer.terms') }}</a></li>
          <li><a :href="`${meta.base_url}/privacy`">{{ $t('footer.privacy') }}</a></li>
          <li><a :href="`${meta.base_url}/help`">{{ $t('footer.help') }}</a></li>
          <li><a :href="`${meta.base_url}/contact`">{{ $t('footer.contact') }}</a></li>
          <li><a :href="`${meta.base_url}/about`">{{ $t('footer.about') }}</a></li>
        </ul>
        <span>Â© 2018 1PAC Vietnam Co.,Ltd</span>
      </div>
    </div>
  </footer>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'app-footer',

  computed: {
    ...mapState('initialStates', [
      'meta'
    ])
  }
}
</script>
