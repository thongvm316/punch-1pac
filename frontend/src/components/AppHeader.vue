<template>
  <header class="header">
    <div class="container grid-xl">
      <div class="navbar">
        <section class="navbar-section">
          <router-link to="/dashboard" class="navbar-brand mr-4">
            <img :src="companyLogoUrl" :alt="companyName">
          </router-link>
          <router-link class="mr-5" to="/dashboard">{{ $t('header.dashboard') }}</router-link>
          <router-link class="mr-5" to="/attendances">{{ $t('header.attendances') }}</router-link>
          <router-link class="mr-5" to="/requests">{{ $t('header.requests') }}</router-link>
          <router-link class="mr-5" to="/groups">{{ $t('header.groups') }}</router-link>
        </section>
        <section class="navbar-section">
          <punch/>
          <header-announcements/>
          <div class="dropdown" :class="{ active: isDropdownActive }" @click="toggleDropdown">
            <a class="dropdown-toggle">
              <div class="tile tile-centered">
                <div class="tile-icon">
                  <img :src="userAvatarUrl" class="avatar avatar-md" :alt="userName">
                </div>
                <div class="tile-content mr-2">{{ userName }}</div>
              </div>
            </a>
            <ul class="menu">
              <li class="menu-item"><router-link to="/settings">{{ $t('header.settings') }}</router-link></li>
              <li class="menu-item"><router-link to="/company/settings">{{ $t('header.companySettings') }}</router-link></li>
              <li class="menu-item"><a href="#" @click="logout($event)">{{ $t('header.logout') }}</a></li>
            </ul>
          </div>
        </section>
      </div>
    </div>
  </header>
</template>

<script>
import axios from 'axios'
import HeaderAnnouncements from '../components/HeaderAnnouncements.vue'
import Punch from '../components/Punch.vue'

export default {
  name: 'app-header',
  data () {
    return {
      isDropdownActive: false
    }
  },

  components: {
    HeaderAnnouncements,
    Punch
  },

  methods: {
    logout () {
      axios.post('/logout').then(() => { window.location.href = '/' })
    },

    toggleDropdown () {
      this.isDropdownActive = !this.isDropdownActive
    }
  },

  computed: {
    companyName () {
      return window.INITIAL_STATE.company.name
    },

    companyLogoUrl () {
      return window.INITIAL_STATE.company.logo_url
    },

    userName () {
      return window.INITIAL_STATE.user.name
    },

    userAvatarUrl () {
      return window.INITIAL_STATE.user.avatar_url
    }
  }
}
</script>
