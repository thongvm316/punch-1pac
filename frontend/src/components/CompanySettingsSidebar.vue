<template>
  <ul class="setting-sidebar">
    <li>
      <a
        href="#profile"
        :class="{ active: activeTabName === 'profile' }"
        @click.prevent="setActive('profile')"
      >
        <p-ico-company-profile />
        {{ $t('sidebar.company.profile') }}
      </a>
    </li>
    <li>
      <a
        href="#users"
        :class="{ active: activeTabName === 'users' }"
        @click.prevent="setActive('users')"
      >
        <p-ico-users />
        {{ $t('sidebar.company.users') }}
      </a>
    </li>
    <li>
      <a
        href="#business-days"
        :class="{ active: activeTabName === 'business-days' }"
        @click.prevent="setActive('business-days')"
      >
        <p-ico-business-day />
        {{ $t('sidebar.company.businessDays') }}
      </a>
    </li>
    <li>
      <a
        href="#holidays"
        :class="{ active: activeTabName === 'holidays' }"
        @click.prevent="setActive('holidays')"
      >
        <p-ico-holiday />
        {{ $t('sidebar.company.holidays') }}
      </a>
    </li>
    <li>
      <a
        href="#allowed-ips"
        :class="{ active: activeTabName === 'allowed-ips' }"
        @click.prevent="setActive('allowed-ips')"
      >
        <p-ico-allowed-ip />
        {{ $t('sidebar.company.allowedIPs') }}
      </a>
    </li>
  </ul>
</template>

<script>
import activeTab from '../mixins/active-tab'
import PIcoCompanyProfile from '@/punch-ui/punch-icons/PIcoCompanyProfile'
import PIcoUsers from '@/punch-ui/punch-icons/PIcoUsers'
import PIcoBusinessDay from '@/punch-ui/punch-icons/PIcoBusinessDay'
import PIcoHoliday from '@/punch-ui/punch-icons/PIcoHoliday'
import PIcoAllowedIp from '@/punch-ui/punch-icons/PIcoAllowedIp'

export default {
  name: 'CompanySettingsSidebar',

  components: {
    PIcoCompanyProfile,
    PIcoUsers,
    PIcoBusinessDay,
    PIcoHoliday,
    PIcoAllowedIp
  },

  mixins: [activeTab]
}
</script>
