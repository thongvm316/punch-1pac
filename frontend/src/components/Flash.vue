<template>
  <transition name="fade">
    <div class="toast toast-success text-center toast-flash"
      :class="{ 'toast-success': type === 'success', 'toast-error': type === 'error' }"
      v-if="message">
      {{ message }}
    </div>
  </transition>
</template>

<script>
import { SET_FLASH_MESSAGE } from '../store/mutation-types'
import { mapState, mapMutations } from 'vuex'

export default {
  name: 'flash',

  computed: {
    ...mapState('flash', ['message', 'type', 'timeout'])
  },

  methods: {
    ...mapMutations('flash', [SET_FLASH_MESSAGE])
  },

  watch: {
    message: function(newMsg) {
      if (newMsg) {
        setTimeout(() => {
          this[SET_FLASH_MESSAGE]({ message: null })
        }, this.timeout)
      }
    }
  }
}
</script>
