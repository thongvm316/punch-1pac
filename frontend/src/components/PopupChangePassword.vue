<template>
  <div class="modal" :class="{ active: open }">
    <span class="modal-overlay" @click="remind"></span>
    <div class="modal-container">
      <div class="modal-header">
        <button ref="btnCloseModal" class="btn btn-clear float-right" @click="remind"></button>
        <h3 class="modal-title">{{ $t('popup.changePassword.title') }}</h3>
      </div>
      <div class="modal-body">
        <div class="content">{{ $t('popup.changePassword.description') }}</div>
      </div>
      <div class="modal-footer">
        <router-link ref="btnChangePassword" class="btn btn-success" to="/settings/password" @click.native="remind">{{ $t('popup.changePassword.btnChange') }}</router-link>
        <button ref="btnCancel" type="button" class="btn" @click="remind">{{ $t('popup.changePassword.btnRemind') }}</button>
      </div>
    </div>
  </div>
</template>

<script>
import { mapMutations } from 'vuex'
import { INITIAL_STATES_UPDATE_PASSWORD_CHANGED } from '../store/mutation-types.js'

export default {
  name: 'popup-change-password',

  data() {
    return {
      open: false
    }
  },

  methods: {
    ...mapMutations('initialStates', [INITIAL_STATES_UPDATE_PASSWORD_CHANGED]),

    remind() {
      this[INITIAL_STATES_UPDATE_PASSWORD_CHANGED](true)
      this.open = !this.currentUser.password_changed
    }
  },

  created() {
    this.open = !this.currentUser.password_changed
  }
}
</script>
