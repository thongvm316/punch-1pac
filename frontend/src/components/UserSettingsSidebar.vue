<template>
  <ul class="setting-sidebar">
    <li>
      <a
        href="#profile"
        :class="{ active: activeTabName === 'profile' }"
        @click.prevent="setActive('profile')"
      >
        <svg
          width="24px"
          height="24px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          fill="currentColor"
          fill-rule="evenodd"
        ><path d="M12,0 C8.94646154,0 6.46153846,2.48492308 6.46153846,5.53846154 C6.46153846,8.592 8.94646154,11.0769231 12,11.0769231 C15.0535385,11.0769231 17.5384615,8.592 17.5384615,5.53846154 C17.5384615,2.48492308 15.0535385,0 12,0 Z M12,9.23076923 C9.96369231,9.23076923 8.30769231,7.57476923 8.30769231,5.53846154 C8.30769231,3.50215385 9.96369231,1.84615385 12,1.84615385 C14.0363077,1.84615385 15.6923077,3.50215385 15.6923077,5.53846154 C15.6923077,7.57476923 14.0363077,9.23076923 12,9.23076923 Z M23.0695385,17.4996923 C23.0658462,17.4332308 23.0473846,17.3723077 23.0307692,17.3095385 C23.016,17.256 23.0086154,17.2006154 22.9846154,17.1507692 C22.9624615,17.1009231 22.9255385,17.0603077 22.8923077,17.0141538 C22.8535385,16.9606154 22.8203077,16.9070769 22.7704615,16.8627692 C22.7593846,16.8535385 22.7556923,16.8406154 22.7446154,16.8313846 C16.5175385,11.6123077 7.47692308,11.6123077 1.24984615,16.8313846 C1.23876923,16.8406154 1.23507692,16.8535385 1.224,16.8627692 C1.176,16.9070769 1.14092308,16.9606154 1.10215385,17.016 C1.07076923,17.0621538 1.03384615,17.1027692 1.01169231,17.1507692 C0.987692308,17.2006154 0.980307692,17.256 0.965538462,17.3095385 C0.948923077,17.3723077 0.930461538,17.4332308 0.926769231,17.4996923 C0.930461538,17.5126154 0.923076923,17.5236923 0.923076923,17.5384615 L0.923076923,22.1538462 C0.923076923,23.1710769 1.752,24 2.76923077,24 L21.2307692,24 C22.248,24 23.0769231,23.1710769 23.0769231,22.1538462 L23.0769231,17.5384615 C23.0769231,17.5236923 23.0695385,17.5126154 23.0695385,17.4996923 Z M21.2307692,22.1538462 L2.76923077,22.1538462 L2.76923077,17.9944615 C8.18584615,13.7021538 15.8141538,13.7021538 21.2307692,17.9944615 L21.2307692,22.1538462 Z" /></svg>
        {{ $t('sidebar.user.profile') }}
      </a>
    </li>
    <li>
      <a
        href="#password"
        :class="{ active: activeTabName === 'password' }"
        @click.prevent="setActive('password')"
      >
        <svg
          width="24px"
          height="24px"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          fill="currentColor"
          fill-rule="evenodd"
        ><path d="M19.9316862,4.07943166 C19.3870244,3.53395674 18.6625638,3.23341866 17.8919491,3.23341866 C17.1214465,3.23341866 16.3969858,3.53395674 15.8523241,4.07931956 C15.3075503,4.62457029 15.0075494,5.34951611 15.0075494,6.12031052 C15.0075494,6.89132913 15.3076623,7.61616285 15.852212,8.16085308 C16.3968738,8.70610381 17.1211104,9.00641769 17.8916131,9.00641769 C18.6620037,9.00641769 19.3864643,8.70621591 19.9312381,8.16118938 C20.476236,7.61616285 20.7763489,6.89132913 20.7763489,6.12031052 C20.7763489,5.34962821 20.476348,4.62468239 19.9316862,4.07943166 Z M18.7434453,6.97192849 C18.515812,7.19960205 18.2132346,7.32492901 17.891501,7.32492901 C17.5699915,7.32492901 17.2678622,7.19971415 17.0402289,6.97192849 C16.8128198,6.74459122 16.6878007,6.44214745 16.6878007,6.12031052 C16.6878007,5.79858569 16.8130438,5.49591773 17.040565,5.26801996 C17.2679742,5.04034639 17.5703276,4.91490734 17.8918371,4.91490734 C18.2133466,4.91490734 18.515588,5.04034639 18.7432212,5.26813206 C18.9706304,5.49580563 19.0958735,5.79847359 19.0958735,6.12031052 C19.0959855,6.44214745 18.9708544,6.74459122 18.7434453,6.97192849 Z M21.7903922,2.21959309 C18.8434829,-0.728841261 14.0489581,-0.729289658 11.1022729,2.21914469 C8.95331215,4.3705534 8.31264561,7.57502243 9.42213354,10.3619218 L6.51353655,13.2722424 L3.85072072,13.2722424 C3.38660435,13.2722424 3.01053903,13.6485595 3.01053903,14.1129867 L3.01053903,16.2850217 L0.840181689,16.2850217 C0.376065324,16.2850217 0,16.6613388 0,17.125766 L0,23.1512125 C0,23.6156397 0.376065324,23.9919569 0.840181689,23.9919569 L3.85083274,23.9919569 C4.0735369,23.9919569 4.28705508,23.9035106 4.44467316,23.7460111 L13.6120636,14.5798801 C16.4080762,15.7144365 19.6264202,15.0793943 21.7905042,12.9138611 C24.7365173,9.96565093 24.7365173,5.16825163 21.7903922,2.21959309 Z M20.6021512,11.7249365 C18.8055067,13.5230084 16.0743561,13.9645673 13.8063136,12.823397 C13.4825636,12.6608531 13.091263,12.7236287 12.8350636,12.9798875 L3.50299752,22.3105803 L1.68036338,22.3105803 L1.68036338,17.9666225 L3.85083274,17.9666225 C4.31483708,17.9666225 4.69101443,17.5903053 4.69101443,17.1258781 L4.69101443,14.9538431 L6.86159582,14.9538431 C7.084412,14.9538431 7.2980422,14.8652847 7.45566029,14.7076732 L11.0183667,11.1428051 C11.2732218,10.8877793 11.3371877,10.4985707 11.1773291,10.1752765 C10.0588792,7.91412264 10.506304,5.19448286 12.2906258,3.40806928 C14.5820814,1.11552762 18.3103596,1.11586392 20.6021512,3.40862978 C22.8931586,5.70161984 22.8931586,9.43217062 20.6021512,11.7249365 Z" /></svg>
        {{ $t('sidebar.user.password') }}
      </a>
    </li>
    <li>
      <a
        href="#security"
        :class="{ active: activeTabName === 'security' }"
        @click.prevent="setActive('security')"
      >
        <svg
          width="22px"
          height="24px"
          viewBox="0 0 22 24"
          version="1.1"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          fill="currentColor"
          fill-rule="evenodd"
        ><path d="M22.0700791,6.17955323 C22.0577426,5.53158895 22.0459669,4.91946713 22.0459669,4.32728267 C22.0459669,3.86334697 21.6694809,3.48722527 21.2048454,3.48722527 C17.6076492,3.48722527 14.8688454,2.45473872 12.5859295,0.237883255 C12.2593501,-0.0793224189 11.7396492,-0.0792104112 11.4131819,0.237883255 C9.13049029,2.45473872 6.39213515,3.48722527 2.79516319,3.48722527 C2.33063982,3.48722527 1.95404169,3.86334697 1.95404169,4.32728267 C1.95404169,4.91957914 1.94237814,5.53192498 1.92992954,6.18000126 C1.81463982,12.2100453 1.65673328,20.4684815 11.7240604,23.9535997 C11.902467,24.0154388 12.0965323,24.0154388 12.2749389,23.9535997 C22.343051,20.4683695 22.1853688,12.2097092 22.0700791,6.17955323 Z M11.9996118,22.2684445 C3.36555571,19.1359265 3.49609777,12.2737776 3.61194823,6.21203545 C3.61890151,5.84823459 3.62563048,5.4956345 3.63034076,5.14997888 C6.99931272,5.00795318 9.6933688,3.98834751 11.9996118,1.98184241 C14.3060791,3.98834751 17.0005837,5.00806518 20.36978,5.14997888 C20.3744903,5.49552249 20.3812193,5.84789857 20.3881725,6.21147541 C20.5039109,12.2734416 20.6343408,19.1358145 11.9996118,22.2684445 Z M14.8960978,9.08178353 L10.8363968,13.1361245 L9.104023,11.4059423 C8.77553702,11.0779839 8.24293889,11.0779839 7.91456506,11.4059423 C7.58607908,11.7341247 7.58607908,12.2659371 7.91456506,12.5940075 L10.2416679,14.9181663 C10.4058548,15.0821455 10.6211819,15.1641351 10.8363968,15.1641351 C11.0516118,15.1641351 11.2669389,15.0821455 11.4311258,14.9181663 L16.0854436,10.2698487 C16.4140417,9.94177829 16.4140417,9.40985394 16.0855557,9.08189553 C15.7571819,8.75382512 15.2245837,8.75371311 14.8960978,9.08178353 Z" /></svg>
        {{ $t('sidebar.user.security') }}
      </a>
    </li>
  </ul>
</template>

<script>
import activeTab from '../mixins/active-tab'

export default {
  name: 'UserSettingsSidebar',

  mixins: [activeTab]
}
</script>
