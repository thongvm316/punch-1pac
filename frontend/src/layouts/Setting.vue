<template>
  <div>
    <app-header></app-header>
    <div class="app-container">
      <app-sidebar></app-sidebar>
      <div class="main">
        <slot></slot>
      </div>
    </div>
  </div>
</template>

<script>
  import AppHeader from '../components/AppHeader.vue'
  import AppSidebar from '../components/AppSidebar.vue'

  export default {
    name: 'setting-layout',
    components: {
      appSidebar: AppSidebar,
      appHeader: AppHeader
    }
  }
</script>
