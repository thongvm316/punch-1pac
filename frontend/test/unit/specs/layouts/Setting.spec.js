import { shallowMount } from '@vue/test-utils'

import wrapperOps from '../../supports/wrapper'

import AppHeader from '@/components/AppHeader.vue'
import AppFooter from '@/components/AppFooter.vue'
import CompanySettingsSidebar from '@/components/CompanySettingsSidebar.vue'
import UserSettingsSidebar from '@/components/UserSettingsSidebar.vue'
import Flash from '@/components/Flash.vue'
import PopupChangePassword from '@/components/PopupChangePassword'
import Setting from '@/layouts/Setting'

const scopedSlots = {
  default: '<p class="default-scoped">Setting page</p>'
}

describe('Setting.vue', () => {
  describe('when sidebarType is company', () => {
    let wrapper

    beforeEach(() => {
      Object.assign(wrapperOps, {
        propsData: {
          sidebarType: 'company',
          subtitle: 'Setting'
        },
        scopedSlots
      })

      wrapper = shallowMount(Setting, wrapperOps)
    })

    afterEach(() => { wrapper.vm.$destroy() })

    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('should have slot', () => {
      expect(wrapper.find('.default-scoped').exists()).toBeTruthy()
    })

    it('should have child components', () => {
      expect(wrapper.find(AppHeader).exists()).toBeTruthy()
      expect(wrapper.find(AppFooter).exists()).toBeTruthy()
      expect(wrapper.find(CompanySettingsSidebar).exists()).toBeTruthy()
      expect(wrapper.find(Flash).exists()).toBeTruthy()
      expect(wrapper.find(PopupChangePassword).exists()).toBeTruthy()
    })
  })

  describe('when sidebarType is company', () => {
    let wrapper

    beforeEach(() => {
      Object.assign(wrapperOps, {
        propsData: {
          sidebarType: 'user',
          subtitle: 'Setting'
        },
        scopedSlots
      })

      wrapper = shallowMount(Setting, wrapperOps)
    })

    afterEach(() => { wrapper.vm.$destroy() })

    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('should have slot', () => {
      expect(wrapper.find('.default-scoped').exists()).toBeTruthy()
    })

    it('should have child components', () => {
      expect(wrapper.find(AppHeader).exists()).toBeTruthy()
      expect(wrapper.find(AppFooter).exists()).toBeTruthy()
      expect(wrapper.find(UserSettingsSidebar).exists()).toBeTruthy()
      expect(wrapper.find(Flash).exists()).toBeTruthy()
      expect(wrapper.find(PopupChangePassword).exists()).toBeTruthy()
    })
  })
})
