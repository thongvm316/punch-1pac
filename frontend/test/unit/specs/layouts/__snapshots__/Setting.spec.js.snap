// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting.vue should render Setting layout correctly 1`] = `
<main>
  <div class="container grid-xl mt-10">
    <div class="columns">
      <div class="column col-3">
        <p class="sidebar-scoped">Setting side bar</p>
      </div>
      <div class="column col-9">
        <div class="box-setting">
          <p class="default-scoped">Setting page</p>
        </div>
      </div>
    </div>
  </div>
</main>
`;
